utensor_cgen/transformer/tflite_exporter.py

```diff
@@ -0,0 +1,412 @@
+
+      #q_param = self.__create_quantization_param([-20],[200]) #TODO: workout how to treat the quantization here
+
+      tflite.Tensor.TensorStart(self.fbuilder)
+      tflite.Tensor.TensorAddShape(self.fbuilder, shape_vec)
+      tflite.Tensor.TensorAddType(self.fbuilder, tensor_type_lookup(tensor_info.dtype))
+      tflite.Tensor.TensorAddName(self.fbuilder, tensor_name)
+      #tflite.Tensor.TensorAddQuantization(self.fbuilder, q_param)
+      tflite.Tensor.TensorAddIsVariable(self.fbuilder, True)
+      #tflite.Tensor.TensorAddIsVariable(self.fbuilder, False) #FIXME: TOCO outputs False here, no idea why
+
+      self.tensor_index[tensor_info.name] = tflite.Tensor.TensorEnd(self.fbuilder)
+
+  def __legalize_name(self, str):
+    #TODO: legalize the name
+    return str
+
+  def output(self):
+    return self.fbuilder.Output()
+
+  def __fb_vector(self, vector_constructor, arr_list):
+    prepend_method_table = dict()
+    prepend_method_table[tflite.SubGraph.SubGraphStartTensorsVector] = self.fbuilder.PrependUOffsetTRelative
+    prepend_method_table[tflite.SubGraph.SubGraphStartOperatorsVector] = self.fbuilder.PrependUOffsetTRelative
+    prepend_method_table[tflite.Model.ModelStartSubgraphsVector] = self.fbuilder.PrependUOffsetTRelative
+    prepend_method_table[tflite.Model.ModelStartBuffersVector] = self.fbuilder.PrependUOffsetTRelative
+    prepend_method_table[tflite.Model.ModelStartOperatorCodesVector] = self.fbuilder.PrependUOffsetTRelative
+    prepend_method_table[tflite.Buffer.BufferStartDataVector] = self.fbuilder.PrependUint8
+    prepend_method_table[tflite.Tensor.TensorStartShapeVector] = self.fbuilder.PrependInt32
+    prepend_method_table[tflite.Operator.OperatorStartOutputsVector] = self.fbuilder.PrependInt32
+    prepend_method_table[tflite.Operator.OperatorStartInputsVector] = self.fbuilder.PrependInt32
+    prepend_method_table[tflite.QuantizationParameters.QuantizationParametersStartScaleVector] = self.fbuilder.PrependFloat32
+    prepend_method_table[tflite.QuantizationParameters.QuantizationParametersStartZeroPointVector] = self.fbuilder.PrependFloat32
+    prepend_method_table[tflite.SubGraph.SubGraphStartOutputsVector] = self.fbuilder.PrependInt32
+    prepend_method_table[tflite.SubGraph.SubGraphStartInputsVector] = self.fbuilder.PrependInt32
+
+    preprend_method = prepend_method_table[vector_constructor]
+
+    num_items = len(arr_list)
+    #assert num_items > 0
+
+    vector_constructor(self.fbuilder, num_items)
+    # reversed to handle prepend
+    for it in reversed(arr_list):
+      preprend_method(it)
+    
+    return self.fbuilder.EndVector(num_items)
+
+  def __tflm_graph_legalize_(self, ugraph):
+    for _, op in ugraph.ops_info.items():
+      assert op.op_type.startswith("TFLM_") or op.op_type in self.static_op_types
+      #TODO: strip the leading TFLM_ only
+      op.op_type = op.op_type.replace("TFLM_", '')
+# How to construct the quantization parameter for intermediate tensors?
```

Comment:
I don't know what you mean by quantization parameters for intermediate tensors

Reply:
This is a misplaced comment meant for the routine that construct the quantization parameters for intermediate tensors. It’s likely to have something to do with offline quantizations.